src/modules/undo-manager.coffee

```diff
@@ -51,8 +51,8 @@ class UndoManager
       timestamp = new Date().getTime()
       if @lastRecorded + @options.delay > timestamp and @stack.undo.length > 0
         change = @stack.undo.pop()
-        undoDelta = undoDelta.compose(change.undo)
-        changeDelta = change.redo.compose(changeDelta)
+        undoDelta = new Delta().compose(undoDelta).compose(change.undo)
```

Comment:
This works but might be more clear to use [`slice()`](https://github.com/ottypes/rich-text#slice). Passing no parameters will create a copy.